src/query/functions/binary/unless.go

```diff
@@ -82,24 +112,16 @@ func distinctLeft(
 		leftSigs[idFunction(meta.Tags)] = idx
 	}
 
-	for _, rs := range rhs {
-		// If there's no matching entry in the left-hand side Vector, add the sample.
+	rhsIndices := make([]int, len(rhs))
```

Comment:
maybe initialize as `make([]int, 0, len(rhs))` and append. Just better to avoid a potential panic.

Reply:
Luckily, was refactoring this stuff already, so it should be fixed :)